src/render/tutorials/openapi-and-swagger-ui/2_0.html.md

```diff
@@ -0,0 +1,721 @@
+
+Let's add to endpoints to the `openapi.yaml` that will serve our `index.html` file and the `openapi.yaml` file itself:
+
+```yaml
+/swagger/spec:
+  get:
+    operationId: openapi-operation
+    responses:
+      default:
+        description: exposes openapi spec file to be consumed by swagger ui
+/swagger/ui:
+  get:
+    operationId: swaggerui-operation
+    responses:
+      default:
+        description: swagger-ui web app
+```
+
+And define the operations in `operations.conf`:
+
+```HOCON
+{
+  operationId = openapi-operation
+  handlers = [
+    {
+      name = static
+      config.webroot = openapi.yaml
+    }
+  ]
+}
+{
+  operationId = swaggerui-operation
+  handlers = [
+    {
+      name = static
+      config.webroot = webroot/swagger/
+    }
+  ]
+}
+```
+
+The first operation serves the `openapi.yaml` file from as a static resource, and the second operation serves the Swagger UI web app. Now we need to define `static` handler. It just serves static resources from a classpath:
+
+```java
+public class StaticHandlerFactory implements RoutingHandlerFactory {
+
+  @Override
+  public String getName() {
+    return "static";
+  }
+
+  @Override
+  public Handler<RoutingContext> create(Vertx vertx, JsonObject config) {
+    return StaticHandler.create(config.getString("webroot"));
+  }
+}
+```
+
+Remember about updating the META-INF.
+
+### Using Swagger UI
+We're done. After you run the application you can go to `http://localhost:8092/swagger/ui` in your browser. It'll download the `openapi.yaml` file from the `/swagger/spec` path, parse it and display the graphical interface.
+
+Top part should look like this:
+
+![Top part of Swagger UI interface](/img/blog/openapi-and-swagger-ui/swagger-ui-top.png)
+
+It shows some metadata we specified in the OpenAPI file. It also allows us to choose a server and its variables that we defined. It's used for testing the endpoints.
+
+The next part shows the list of all endpoints and their methods:
+
+![Swagger UI endpoints](/img/blog/openapi-and-swagger-ui/swagger-ui-endpoints.png)
+
+We can see `summary` is displayed here as well. We can also clearly see that `POST /api/user` is deprecated. Let's click on the `GET /api/user/isadmin` endpoint. You should see all the information abouyt this endpoint. It consumes one required query parameter `datatype` that can either be `json` or `xml`. It can also return one of those two content types. Example responses are also provided. You can click on the `Try it out` button, specify the query parameter and `Execute`. Swagger UI will call this endpoint and show you the response. Curl commands are also provided. Swagger UI can handle many things, such as image uploading or even different types authorization. You can read more about it [here](https://swagger.io/tools/swagger-ui/).

```

Comment:
```suggestion
We can see `summary` is displayed here as well. We can also clearly see that `POST /api/user` is deprecated. Let's click on the `GET /api/user/isadmin` endpoint. You should see all the information about this endpoint. It consumes one required query parameter `datatype` that can either be `json` or `xml`. It can also return one of those two content types. Example responses are also provided. You can click on the `Try it out` button, specify the query parameter and `Execute`. Swagger UI will call this endpoint and show you the response. Curl commands are also provided. Swagger UI can handle many things, such as image uploading or even different types of authorization. You can read more about it [here](https://swagger.io/tools/swagger-ui/).
```